### PR TITLE
[12.0][REF] Make search engine locomotive export settings generic 

### PR DESCRIPTION
**Pull request by simahawk**
**Implements:** https://github.com/shopinvader/odoo-shopinvader/issues/466
**Requires:** https://github.com/OCA/search-engine/pull/35
**Causes:** https://github.com/shopinvader/odoo-shopinvader-payment/pull/15

### CODE REVIEW

File: shopinvader_locomotive/component/shopinvader_site_export_mapper.py [hunk 1]
```diff
@@ -73,6 +76,89 @@ def erp_config(self, record):
...
+                [
+                    "*",
+                    {
+                        "name": "category",
+                        "template_handle": "category",
+                        "index": "categories",
+                    },
+                ]
+            )
+        index_product = se_backend.index_ids.filtered(
```
Comment (rousseldenis) [inline shopinvader_locomotive/component/shopinvader_site_export_mapper.py]: Prefer using :

```suggestion
        from odoo.fields import first
        index_product = first(se_backend.index_ids.filtered(
```
so you can remove 
``` 
index_product = index_product[0]
```

**Review by lmignon — CHANGES_REQUESTED** [review]
@simahawk IMO OCA/search-engine#35 should not be required. `_search_engine_name`could be defined as a simple computed fields.Char on  `se.backend.spec.abstract`:

```python
search_engine_name = fields.Char(compute="_compute_search_engine_name")

@api.depends("specific_model")
def _compute_search_engine_name(self):
    for rec in self:
        rec._search_engine_name = rec._specific_model.split(".")[-1]

```

Since this field will only be used by shopinvader (at least for the moment) it should be declared ont shopinvader_search_engine.

What do you think about?

**Comment by simahawk** [comment]
@lmignon I thought about it but I don't like it because it's not explicit and potentially you can have whatever string at the end of the model name. However, if you prefer it like this, no problem :)

> Since this field will only be used by shopinvader (at least for the moment) it should be declared ont shopinvader_search_engine.

This I don't agree. In the core module there should be something to say "this is XYZ search engine".
So, in any case I'd put it there rather than here.

**Comment by lmignon** [comment]
@simahawk My concern is that this new class attribute is somehow redundant with the model name. I understand the need to be able to say "this is XYZ search engine".  But IMO, we already have 2 places where this needs is already covered. The attribute `_name` of the specialized backend and the field `specific_model` on "se.backend".

You're right to prefer explicit over implicit to implement this UC. So I'll not block since at this stage it's a matter of feeling. 

If it's ok for you, I'll merge https://github.com/OCA/search-engine/pull/35

**Comment by simahawk** [comment]
@lmignon the good thing is that the attribute name comes from a property -> we can convert it to a field if we think it's better. It won't change current implementation -> let's merge it and improve later, fine?

**Comment by lmignon** [comment]
It's fine for me..

**Review by lmignon — APPROVED** [review]

**Comment by lmignon** [comment]
@simahawk Can you fix flake8 PLZ...

**Comment by simahawk** [comment]
Linting fixed. 2 builds are failing tho but they seem unrelated:

https://travis-ci.org/shopinvader/odoo-shopinvader/jobs/603896021
https://travis-ci.org/shopinvader/odoo-shopinvader/jobs/603896022

Seems there's lack of permissions somewhere. Do they ring any bell to you?

**Comment by simahawk** [comment]
hmm might be related to the change in the base class for tests. Debugging in WIP

**Comment by simahawk** [comment]
I finally found the reason for test failure: bad isolation between demo data for tests. `shopinvader_elasticsearch` adds a SE to base demo backend. With all modules installed this causes SE settings to be included into basic site export.
Should be green now.

**Comment by simahawk** [comment]
external-content duckduckgo com

**Comment by simahawk** [comment]
@lmignon finally greeen :)
I guess I can try my new super powers now! (thanks, BTW :wink: )

**Pull request merged**